### PR TITLE
donut-v9 theme exploration

### PR DESCRIPTION
**Pull request by agarwalpratham**
## Feature Added

- Chart Migrated from Class Component to Functional Component.
- Empty Chart feature added using hooks.
- components styled using v9 design tokens.
- Introduced midValueId in pie using getId utility since getClassNames is no longer used.
- Modified insideDonutString test with getById query.

## Issues

- Legend buttons are rendered from common components, so legend text style overrides root style.

## Future Work

- Common Components styling using design tokens.
- If color is undefined by user, it should obtain color from utility package and same color needs to be mapped to both legend and its corresponding rendered arcs.

### CODE REVIEW

File: packages/react-charting/src/components/DonutChart/DonutChart.styles.ts [hunk 1]
```diff
@@ -1,31 +1,22 @@
...
-    },
-    legendContainer: {
-      paddingTop: '16px',
-      width: `${width}px`,
-    },
-  };
-};
+export const useStyles = makeStyles({
+  root: {
+    text: typographyStyles.body1,
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/DonutChart/DonutChart.styles.ts:5]: which font are you using?

File: packages/react-charting/src/components/DonutChart/Pie/Pie.styles.ts [hunk 2]
```diff
@@ -1,19 +1,10 @@
...
-      },
-    },
-  };
-};
+export const useStyles = makeStyles({
+  root: {},
+  insideDonutString: {
+    fontSize: tokens.fontSizeBase600,
+    fontWeight: tokens.fontWeightSemibold,
+    fill: tokens.colorNeutralForeground1Selected,
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/DonutChart/Pie/Pie.styles.ts:8]: how do you handle high contrast mode?

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/36aacc5ff33fdb056c46a439da81a5c9b274e8ee'>36aacc5ff33fdb056c46a439da81a5c9b274e8ee</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by AtishayMsft** [comment]
The exploration for donut chart in V9 has concluded. Closing this PR.

**Pull request closed without merging**